### PR TITLE
run more checks in golangci-lint

### PR DESCRIPTION
**Pull request by briskt**

### CODE REVIEW

File: .golangci.yaml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+run:
+    timeout: 2m
+linters:
```
Comment (mtompset) [inline .golangci.yaml:3]: That is a LOT of linters!

File: functional_test.go [hunk 2]
```diff
@@ -87,7 +87,7 @@ func sendRequest(url string, c *http.Cookie) error {
 func weSendARequestWithAuthorizationDataAuthorizingAccess(where, level string) error {
 	var c *http.Cookie
 	url := p.Host
-	expires := time.Now().Add(1000 + time.Second*time.Duration(rand.Intn(1000)))
+	expires := time.Now().Add(1000 + time.Second*time.Duration(rand.Intn(1000))) // #nosec G404 weak random number is OK
```
Comment (jason-jackson) [inline functional_test.go:90]: Not looked much at it, but off the top of my head, do we even need to add a random number of seconds here?

File: main.go [hunk 3]
```diff
@@ -356,6 +356,7 @@ func (p Proxy) getTokenFromAPI(ipAddress string) string {
 		return ""
 	}
 
+	defer resp.Body.Close()
 	token, err := io.ReadAll(resp.Body)
```
Comment (jason-jackson) [inline main.go:360]: What are your thoughts about using resp.Body.Read() instead of io package directly?

**Review by jason-jackson — APPROVED** [review]

File: .golangci.yaml (hunk 1 above)
Comment (briskt) [reply .golangci.yaml:3]: Yeah, it's quite a collection. Most are very narrowly focused.

File: functional_test.go (hunk 2 above)
Comment (briskt) [reply functional_test.go:90]: Not really. I suppose it's a poor man's fuzzing.

File: main.go (hunk 3 above)
Comment (briskt) [reply main.go:360]: It's more complex. What would be the advantage?

Comment (briskt) [reply main.go:360]: I don't give any thought to standard library imports. Should I?

File: functional_test.go (hunk 2 above)
Comment (jason-jackson) [reply functional_test.go:90]: Are you fine with removing it? I think it just adds unnecessary complexity (and complains about a possible security issue).

Comment (briskt) [reply functional_test.go:90]: I guess so. I'm sorry I didn't make that change before merging.

**Pull request merged**